### PR TITLE
 [SHACK-377] Set update channel button

### PR DESCRIPTION
**Pull request by jonsmorrow**
This change makes the update channel button on the
about dialog work. It introduces a new app manged
config file where the selected channel is persisted.
The button is disabled if the user has set a channel
in the user config file and that value will be used.

Signed-off-by: Jon Morrow <jmorrow@chef.io>

### CODE REVIEW

**Comment by jonsmorrow** [comment]
this pr is dependent on pr #51

File: src/about_dialog.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const { BrowserWindow } = require('electron');
+const path = require('path');
+
+const width = process.platform === 'darwin' ? 510 : 530;
+const height = process.platform === 'darwin' ? 325 : 330;
```
Comment (marcparadise) [inline src/about_dialog.js]: I think making this 330 for all platforms will probably look better - windows and linux both have a scroll bar on the About dialog currently.

File: src/about.js [hunk 2]
```diff
@@ -1,35 +1,9 @@
-const { shell, BrowserWindow } = require('electron');
+const { shell } = require('electron');
```
Comment (marcparadise) [inline src/about.js]: For longer-term talk: I'm not clear why we have `about.js` vs `about_dialog.js`.  Can we merge the two?

File: src/chef_workstation.js [hunk 3]
```diff
@@ -117,6 +124,32 @@ function getUserConfig() {
...
+      appConfigCache = {};
+    }
+  }
+  return appConfigCache;
+}
+
+function saveAppConfig() {
+  if ((loadedAppConfig != null && appConfigCache != null) || (loadedAppConfig == null && appConfigCache != {})) {
+    try {
+      if (!fs.existsSync(ws_dir)) {
```
Comment (marcparadise) [inline src/chef_workstation.js]: Good catch - it's likely that the App will be run before `chef-run` does any setup.

File: src/about.js (hunk 2 above)
Comment (jonsmorrow) [reply src/about.js]: The idea of two files is to separate what runs in the render process from the main process. It's easy to get conflicts and think you are spanning them so I tend to keep them separate.

File: src/about_dialog.js (hunk 1 above)
Comment (jonsmorrow) [reply src/about_dialog.js]: I kept it the same that it was figuring the style changes come later.

File: src/chef_workstation.js (hunk 3 above)
Comment (jonsmorrow) [reply src/chef_workstation.js]: yep ^ that.

**Pull request merged**